system/jlib/jmetrics.cpp

```diff
@@ -88,6 +87,90 @@ Bucket &HistogramMetric::findBucket(__uint64 measurement)
 }
 
 
+ScaledTimeFromCyclesHistogram::ScaledTimeFromCyclesHistogram(const char *name, const char *desc, TimeUnits _timeUnits, const std::vector<BucketDef> &bucketDefs, const MetricMetaData &_metaData) :
```

Comment:
The code is broadly good, but it should share almost all of the implementation with the normal histogram - not duplicate it.  

I expected a derived class with a scaling factor say `double scaleMeasureToReport` which multiplies the sum when reporting, and the limits are divided by it.  That would then modify two lines of the original code, and I would be tempted to merge it into a single class (debatable).

Reply:
@ghalliday 
I debated creating a separate class vs a sub class of the current histogram. What drove me to a separate class was the type, cycle_t in this case, used to record measurements. Using the histogram record measurement method required a conversion from essentially int to uint since cycle_t is signed.

If you are OK with using  cycle_t for recording measurement through the existing record measurement method (with the implicit conversion), then I can simplify the classes into a single class and add a _setScaleFactor_ method that converts the limits and scales the accumulated sum when requested.

Please let me know your thoughts.

Reply:
Hmm it is actually a bug that cycle_t is signed (I just created HPCC-28135).  The histogram should use a _unit64 for all measurements.

Reply:
Since cycle_t is to be unsigned. I refactored into a single class. Note that while cycle_t may be used now, it will be converted to unsigned with a warning in code. Once HPCC-28135 is merged, those issues will disappear.